### PR TITLE
Add a validation check for 413 returned from server and not categorised as an error

### PR DESCRIPTION
**Pull request by AlexPinhasov**
#### What does this PR address?
- [ ] GitHub issue (Add reference - #XX)
- [ ] Refactoring
- [ ] New feature
- [x] Bug fix
- [ ] Adds more tests

#### Are tests included?
- [ ] Yes
- [x] No

#### Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I ran the full test suite before pushing the changes and all the tests pass.

### CODE REVIEW

**Comment by AlexPinhasov** [comment]
@const-cloudinary 
Screenshot 2021-11-29 at 22 49 03

File: Cloudinary/Classes/Core/BaseNetwork/CLDNResponseSerialization.swift [hunk 1]
```diff
@@ -305,6 +313,10 @@ extension CLDNRequest {
         guard let validData = data, validData.count > 0 else {
             return .failure(CLDNError.responseSerializationFailed(reason: .inputDataNilOrZeroLength))
         }
+        
+        if let statusCode = response?.statusCode, HTTPStatusCode(rawValue: statusCode) == .payloadTooLarge {
```
Comment (const-cloudinary) [inline Cloudinary/Classes/Core/BaseNetwork/CLDNResponseSerialization.swift:317]: Please move to `catch` block

Comment (AlexPinhasov) [reply Cloudinary/Classes/Core/BaseNetwork/CLDNResponseSerialization.swift]: Done

File: Cloudinary/Classes/Core/BaseNetwork/CLDNResponseSerialization.swift [hunk 2]
```diff
@@ -161,9 +161,13 @@ extension CLDNRequest {
...
 
         guard let validData = data else {
+            if let statusCode = response?.statusCode,
+                let httpStatusCode = HTTPStatusCode(rawValue: statusCode), httpStatusCode.isError {
+                return .failure(CLDError.error(code: statusCode, userInfo: ["message": httpStatusCode.localizedReason]))
+            }
             return .failure(CLDNError.responseSerializationFailed(reason: .inputDataNil))
         }
-
+
```
Comment (const-cloudinary) [inline Cloudinary/Classes/Core/BaseNetwork/CLDNResponseSerialization.swift:170]: Please remove redundant tabs

**Pull request closed without merging**